draft-ietf-quic-qlog-quic-events.md

```diff
@@ -881,14 +881,17 @@ QUICDatagramsSent = {
 {: #quic-datagramssent-def title="QUICDatagramsSent definition"}
 
 Since QUIC implementations rarely control UDP logic directly, the raw data
-excludes UDP-level headers in all fields.
-
-The `datagram_id` is a qlog-specific concept to allow tracking of QUIC packet
-coalescing inside UDP datagrams. Since QUIC generates many UDP datagrams, unique
-identifiers are required to be able to track them individually in qlog traces.
-However, neither UDP nor QUIC exchanges datagram identifiers on the wire.
-Selecting identifier values is thus left to qlog implementations, which should
-consider how to generate unique values within the scope of their created traces.
+excludes UDP-level headers in all RawInfo fields.
+
+Multiple QUIC packets can be coalesced in a single UDP datagram, especially
+during the handshake (see {{Section 12.2 of QUIC-TRANSPORT}}). However, neither
+QUIC nor UDP themselves provide an explicit mechanism to track this behaviour.
+Implementations willing to track coalescing across packet-level and
+datagram-level qlog events SHOULD thus use the local and qlog-specific concept
+of a "datagram identifier" in the `datagram_id` field. Selecting specific and
+locally unique `datagram_id` values is left to qlog implementations, but in
+general multiple packet-level events sharing the same `datagram_id` SHOULD
+indicate they were coalesced in the same UDP datagram.
```

Comment:
So I might be misreading what you intend but with current text but I could even go further with something like

```suggestion
qlog defines it's own mechanism for tracking coalescing across packet-level and
datagram-level qlog events, a "datagram identifier" carried in `datagram_id` fields.
qlog implementations that want to track coalescing can use this mechanism, the 
selectiion of specific and locally-unique `datagram_id` values is an implementation
choice.
```

Reply:
I feel that loses a (potentially obvious, but still good to call out) aspect in how to use the `datagram_id` though... so maybe something like this then:

```suggestion
qlog defines its own mechanism for tracking coalescing across packet-level and
datagram-level qlog events, a "datagram identifier" carried in `datagram_id`
fields. qlog implementations that want to track coalescing can use this
mechanism, where multiple packet-level events sharing the same `datagram_id`
indicate they were coalesced in the same UDP datagram. The selection of specific
and locally-unique `datagram_id` values is an implementation choice. 
```

Reply:
Looks good